packages/app-accounts/src/UploadButton.tsx

```diff
@@ -0,0 +1,259 @@
+    this.setState({ password: '' }, () => {
+      try {
+        if (!json || !Object.keys(json).length) {
+          return;
+        }
+
+        const pairRestored = keyring.restoreAccount(json, password);
+
+        if (pairRestored) {
+          this.hidePasswordModal();
+          onChangeAccount(pairRestored.publicKey());
+
+          return;
+        } else {
+          this.setState({ error: { key: t('error'), value: t('Unable to upload account into memory') } });
+
+          return;
+        }
+      } catch (e) {
+        console.error('Error processing uploaded file to local storage: ', e);
+        this.setState({ error: { key: t('error'), value: t('Unable to upload account into memory') } });
+
+        return;
+      }
+    });
+
+    return;
+  }
+
+  showPasswordModal = (): void => {
+    this.setState({ isPasswordModalOpen: true });
+  }
+
+  hidePasswordModal = (): void => {
+    this.setState({ isPasswordModalOpen: false });
+  }
+
+  render () {
+    const { uploadedFileKeyringPair, isPasswordModalOpen } = this.state;
+    const { className, icon = 'upload', isCircular = true, isPrimary = true, size = 'tiny', style } = this.props;
+    const address = uploadedFileKeyringPair && uploadedFileKeyringPair.address;
+
+    return (
+      <div className={'accounts--Address-upload'}>
+        { address && isPasswordModalOpen ? (
+            <Modal
+              dimmer='inverted'
+              open={isPasswordModalOpen}
+              onClose={this.hidePasswordModal}
+              size={'mini'}
+            >
+              <Modal.Content>
+                <div className='ui--grid'>
+                  <div className={'accounts--Address-modal'}>
+                    <AddressMini isShort={true} value={address} />
+                    <div className='accounts--Address-modal-message expanded'>
+                      <p>
+                        <Trans i18nKey='unlock.info'>
+                          Please enter your account password to upload and restore it encrypted.
+                        </Trans>
+                      </p>
+                    </div>
+                    {this.renderContent()}
+                  </div>
+                  {this.renderButtons()}
+                </div>
+              </Modal.Content>
+            </Modal>
+          ) : null
+        }
+        <ReactFileReader
+          fileTypes={['.json']}
+          base64={false}
+          multipleFiles={false}
+          handleFiles={this.handleUploadedFiles}
+        >
+          <Button
+            className={className}
+            icon={icon}
+            isCircular={isCircular}
+            isPrimary={isPrimary}
+            size={size}
+            style={style}
+          />
+        </ReactFileReader>
+      </div>
+    );
+  }
+
+  renderContent () {
+    const { error, password, uploadedFileKeyringPair } = this.state;
+    const { t } = this.props;
+    let translateError: InputErrorMessage | undefined = undefined;
```

Comment:
`let translateError: InputErrorMessage;` is the same and preferrable, no need to `= undefined`